### PR TITLE
Runtime Block Sizes

### PR DESCRIPTION
**Pull request by MrBurmark**
# Add support for running multiple block size "tunings" at runtime

This builds on #197 and #210 to add block sizes that can be set at configure time and can be run together in a single run of the perf suite. This is done by adding a layer beneath variants called tunings that can differ for each kernel. By default the perf suite runs all of the tunings but you can specify to run only a subset. Various outputs are changed to output the extra sometimes sparse information.

- This PR is a feature
- It does the following:
  - Modifies/refactors Executor and KernelBase to add support for tunings.
  - Adds block size support at the request of various people

### CODE REVIEW

**Comment by MrBurmark** [comment]
@rhornung67 The gitlab test failure seems to have built and run successfully then encountered a problem after that. Have you seen this kind of error before?

**Comment by rhornung67** [comment]
@MrBurmark yes, on several projects. I found out a few minutes ago that the Gitlab instances are being patched today. That's probably the reason. They are aware of our issues related to this.

**Comment by MrBurmark** [comment]
I think this is reasonably complete now.
I added tunings to the readme so there is some documentation on what tunings are and how to add tunings.
This PR changes just about every kernel in the perf suite so I recommend looking at one kernel and then looking at the common infrastructure files.

File: src/algorithm/SORT-Cuda.cpp [hunk 1]
```diff
@@ -35,7 +35,7 @@ namespace algorithm
   deallocCudaDeviceData(x);
 
 
-void SORT::runCudaVariant(VariantID vid)
+void SORT::runCudaVariant(VariantID vid, size_t /*tune_idx*/)
```
Comment (rchen20) [inline src/algorithm/SORT-Cuda.cpp:38]: This `size_t` looks unused; do we want to remove it?

**Review by artv3 — DISMISSED** [review]

**Comment by rhornung67** [comment]
@MrBurmark I think we should add the blocksize for the default case in the output.

**Comment by MrBurmark** [comment]
I added the block size in the tuning name everywhere that it is known. The only kernels still using default are the sort kernels where cub/rocprim choose the block size and we do not know what it is.

**Review by rhornung67 — APPROVED** [review]

**Pull request merged**